### ISSUE TITLE
False Positive for `extension` blocks

### ISSUE THREAD

**Issue by viceroypenguin** [meziantou/Meziantou.Analyzer#1052]
### Version of the Meziantou.Analyzer NuGet package

3.0.19

### Rule Identifier

MA0068

### Target Framework

net462; net8.0; net10.0

### C# Language version

C# 14

### Description

MA0068 triggers in an `extension()` block when an NRT attribute references the `extension` block parameter; however, it shouldn't, since the parameter exists.

### Reproduction Steps

Minimal code:

```c#
extension(object? obj)
{
	[return: NotNullIfNotNull(nameof(obj))]
	public object? DoSomething() => obj;
}
```

This is valid NRT code, and callers will correctly connect the nullability of the parameter `obj` to the return value of `DoSomething()`. However, MA0068 triggers because `obj` is not a direct parameter to `DoSomething()`. The analyzer should be updated to recognize the extension block parameter as well.

**Comment by viceroypenguin** [comment]
@meziantou thanks!
